dkist/dataset/tiled_dataset.py

```diff
@@ -167,7 +169,26 @@ def _get_axislabels(ax):
                 ylabel = coord.get_axislabel() or coord._get_default_axislabel()
         return (xlabel, ylabel)
 
-    def plot(self, slice_index, share_zscale=False, fig=None, **kwargs):
+    @staticmethod
+    def _ensure_wcs_ordered_axis_lims(ax: WCSAxes) -> None:
+        """
+        Adjust axis limits so the WCS values go from smaller numbers in the bottom left to higher numbers in the upper right.
+        """
+        xmin, xmax = ax.get_xlim()
+        ymin, ymax = ax.get_ylim()
+
+        world_xmin, world_ymin = ax.wcs.pixel_to_world_values(xmin, ymin)
+        world_xmax, world_ymax = ax.wcs.pixel_to_world_values(xmax, ymax)
+
+        new_xmin, new_ymin = ax.wcs.world_to_pixel_values(min(world_xmin, world_xmax), min(world_ymin, world_ymax))
+        new_xmax, new_ymax = ax.wcs.world_to_pixel_values(max(world_xmin, world_xmax), max(world_ymin, world_ymax))
+
+        ax.set_xlim(new_xmin, new_xmax)
+        ax.set_ylim(new_ymin, new_ymax)
```

Comment:
Alright, after some deliberation:
```python
from astropy.wcs.utils import local_partial_pixel_derivatives
```
```suggestion
    local_derivs = local_partial_pixel_derivatives(ax.wcs, *[0]*ax.wcs.pixel_n_dim)
    
    _ = ax.get_xlabel()  # triggers a call to ax._update_tick_and_label_positions()
    
    for coord in ax.coords:
        if 'b' in coord.get_ticklabel_position():
            if local_derivs[coord.coord_index, 0] < 0:
                ax.invert_xaxis()
    
        if 'l' in coord.get_ticklabel_position():
            if local_derivs[coord.coord_index, 1] < 0:
                ax.invert_yaxis()

```

The main things this does differently is: that it does not rely on limits as inputs, which may be unreliable if the image is rotated; uses `invert_[xy]axis` rather than setting limits so the limits wont be affected by numerical error in the round trip transform; and uses `local_partial_pixel_derivatives` to calculate the direction of change of each coordinate.

Reply:
Very cool idea, but ran into some terrible nastiness trying to get it working for `astropy < 7.x`. Solution is just to expose an option to `.plot` to swap one or both of the axes. Let the user just say what they want.